Strip down

marcelmoreau opened the pull request:
I won't capture everything here but I'll try to hit what I can:
- Header
  - Removed nav.secondary
  - Added an image-based logo, utility nav, search form
- Torso
  - Added hero with centered conent (vert and horiz)
  - removed a lot of type/adjacent header stuff
  - removed some deprecated form styles and elements like `input type="button"`
  - removed `articles` and `asides`
  - replaced `div.main` with `main` element
  - increased width of site to 1200 from 1024 per all of our sites
- Footer
  - added social icons via fontawesome
- Misc
  - Added Imarc colors
  - I know I am mixing my `rem` and `em` margins. We'll figure this out
  - I comb'd the CSS
  - I removed Open Sans...simpler and leaner

**What's needed next**
- From our [base component list](https://docs.google.com/document/d/1EI5JUZyWydwbVVjb_rZSVskj2-pbTrkL4iEGYZ0NfHI/edit#) we need:
  - mobile navigation that shows all navs and search
  - call to action element
  - features
  - dedicated iconography/fontawesome section

Commit:
stripping down boilerplate

Commit:
removing input type=button

mbfitz commented:
This all sounds good to me! +1

TommyChanthaboune commented on css/styles.scss:
```diff
@@ -672,281 +611,289 @@ textarea {
+                padding: 0.3rem 0.8rem;
+            }
+            .dropdown li a:hover {
+                background-color: lighten($plain-text, 10%);
```
nice

TommyChanthaboune commented on css/styles.scss:
```diff
@@ -1,90 +1,69 @@
-/**
+ /**
  * Built using Imarc Boilerplate v2.0
```
We should update the version

TommyChanthaboune commented:
+1

P.S.
Great pull request message.

I try to go into detail in my pull request message about what my branch is trying to achieve so I can give my reviewer a high level overview so it gives them context when looking at code.

khamer commented:
- I didn't think we were going to change the look yet; I think there's flat out opposition to including all the Imarc colors like this.
- I think blowing out the reset to one line per selector makes Boilerplate worse. I understand it goes against our formatting guidelines, but I think the reset is a perfect example of when to make an exception.
- Do we really want to to drop all headings from this? I would have expected headings to be part of this for sure, even if we're ditching adjacent headings.
- Personally, I expected aside.primary and the nav.supplemental to disappear from the HTML, but maybe we use those enough to merit them? I didn't think we did.
- copyright and nav in the footer probably makes sense to include by default; think come up last meeting.
- A lot of the aria roles we should drop; any time we're assigning an aria role that is the default for that element, I believe we're not supposed to do that - https://www.w3.org/TR/html-aria/#docconformance - I think that every place we're putting in a role we shouldn't be.
- We should remove js/bp.js if the site isn't going to use it anymore.
- "Gibhub"

Lastly, I think every element we reset so it doesn't have a reasonable default we should restyle; this includes things like `abbr`, `b`, `code`, `i`, `pre`, `sub`, `sup`, etc. If I'm the only one, so be it.

mbfitz commented:
- I'm totally fine with changing the look, and using Imarc colors. It is ours, afterall, and we want it to look good, and like we made it, so I like the consistency
- I must've scrolled right past the reset stuff - agree with Kevin here, I think this is an exception to the rule
- I think we use aside.primary enough to keep it in
- Agreed w/ copyright and nav in the footer

TommyChanthaboune commented:
> I didn't think we were going to change the look yet; I think there's flat out opposition to including all the Imarc colors like this.

I don't mind; It looks great.

> I think blowing out the reset to one line per selector makes Boilerplate worse. I understand it goes against our formatting guidelines, but I think the reset is a perfect example of when to make an exception.

This is a result of csscomb - we need to figure out if csscomb can handle making exceptions for this.

khamer commented:
I think there was creative concern about making every site look like Imarc.com to start, which is why we didn't use Fjord, avoided using the full Imarc palette, used Open Sans and kept Boilerplate's appearance distinct from Imarc's.

Plus... with all the new brand guidelines, I feel like using 'a little bit of Imarc's brand' but not all of it (for example, Gibson/Hind) is a no-no.

keyesatimarc commented:
I was under the impression that the goal of boilerplate was to be a style free starting place. Not sure why we added back all of the Imarc styles. It looks fine as others have said, but goes against the goal of the whole thing. Vanilla.

ghost commented:
> Thanks for doing this Marcel,
- Great write-up on the pull request, I agree with what's needed next.
- I think the nav bar looks a lot closer to the normal setup, including
  the search.
- Overall it's looking a lot cleaner

I didn't think we were going to change the look yet; I think there's flat

> out opposition to including all the Imarc colors like this.

I also was under the impression that Boilerplate shouldn't be Imarc
branded. We talked about Gibson not being used on it because it should be
branding neutral, and I would've expected that to carry through. This is
probably a topic that should get hashed out in the next meeting. I think it
looks a lot better this way, but maybe we can come up with an equally
filled-in boilerplate that uses a more stock appearance? Maybe we should
have creative in on this discussion since branding usage is involved.

> Do we really want to to drop all headings from this? I would have expected
> headings to be part of this for sure, even if we're ditching adjacent
> headings.

I agree that the base Headers should be included, and the adjacent ones
dropped.

copyright and nav in the footer probably makes sense to include by default;

> think come up last meeting.

Agreed - ideally I think this should be set to pull the current year so
that it's one less detail to update.

Personally, I expected aside.primary and the nav.supplemental to disappear

> from the HTML, but maybe we use those enough to merit them? I didn't think
> we did.

I've been seeing these used a lot, but my ratio of 'old site' work versus
'new site' work tends to lean towards 'old site'  -- are multiple site navs
a common occurrence? Is it trivially easy to add either when you need them?

## 

[image: iMarc Logo]

_Shawna O'Neal_ • Jr. Web Engineer • Imarc, A Digital Agency
shawna@imarc.com victoria@imarc.com • 978.462.8848

marcelmoreau commented:
- Design: If we are not adding the Imarc colors, then we are adding other colors. This does not seem like a big deal to me. We are not using the Gibson anywhere. If we replace the hero image with something else, it's even further removed. This is pretty vanilla in my opinion and a decent starting point. I consider this pretty "style free". If it's that big a deal we can change 5 hex values.
- headlines were left out by accident
- we can make those selectors be on one line. I had to remove our mixins in order to comb the file because comb could not process them so we can also remove this reset the next time someone combs.
- we still often have a sidebar on many sites so I left in the aside and nav.supplementary

khamer commented:
So, I think the actions items are
- Swap out the Imarc colors
- Make the reset selectors a single line
- Add headings back in
- Add copyright and nav to footer
- Keep aside.primary and nav.supplemental
- Remove bp.js
- Gibhub
- Increase the version number. If we're using semver, it should be 3.0, but I think we should go with 2.1 instead, and let 3.0 be when we refactor it into components. (Alternatively, we could keep this on a 'develop' branch instead of gh-pages so we can keep working without changing the version and give it a new version when we're ready; if people want to do that, let me know and I can help do it, but if no one else says something I'm going to assume we don't want that.)

Once the above is done on the strip-down branch, I'm cool with merging it in. The following couple things are still undecided, but I don't think block merging this in:
- should be remove the aria roles
- should we 1) restyle or 2) not reset tags like `abbr`, `b`, `code`, `i`, `pre`, `sub`, and `sup`.

ghost commented:
I like the idea of keeping it on a branch, and releasing it with components
for a 3.0 version.

On Fri, Apr 22, 2016 at 1:42 PM, Kevin Hamer notifications@github.com
wrote:

> So, I think the actions items are
> - Swap out the Imarc colors
> - Make the reset selectors a single line
> - Add headings back in
> - Add copyright and nav to footer
> - Keep aside.primary and nav.supplemental
> - Remove bp.js
> - Gibhub
> - Increase the version number. If we're using semver, it should be
>   3.0, but I think we should go with 2.1 instead, and let 3.0 be when we
>   refactor it into components. (Alternatively, we could keep this on a
>   'develop' branch instead of gh-pages so we can keep working without
>   changing the version and give it a new version when we're ready; if people
>   want to do that, let me know and I can help do it, but if no one else says
>   something I'm going to assume we don't want that.)
> 
> Once the above is done on the strip-down branch, I'm cool with merging it
> in. The following couple things are still undecided, but I don't think
> block merging this in:
> - should be remove the aria roles
> - should we 1) restyle or 2) not reset tags like abbr, b, code, i, pre,
>   sub, and sup.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> https://github.com/imarc/boilerplate/pull/42#issuecomment-213525228

## 

[image: iMarc Logo]

_Shawna O'Neal_ • Jr. Web Engineer • Imarc, A Digital Agency
shawna@imarc.com victoria@imarc.com • 978.462.8848

Commit:
cosmetics edits

put in headlines; re-inserted grayscale color scheme; added nav and
copyright to footer;

marcelmoreau commented:
- Colors are back to grayscale
- reset selectors are on one line
- added headings are now in; added copyright and nav to footer
- removed ARIA roles, removed bp.js
- fixed gibhub
- increased version in source code

I'd like to keep `abbr`, `code`, `b`, etc in the reset but unstyled. We often do not use these and they take up space in the sheet. It annoys people. However, if someone used one of these, I would like the box-sizing styles to already be present.

khamer commented:
The only thing I see is that a 'style.css' file that just contains an error message snuck in. Nuke that and we should merge it in.

Looking good Marcel.

mbfitz commented:
Agreed with Kevin. Looks great!!

ghost commented:
Looks good to me, thanks Marcel!

jeffturcotte commented:
👍 Thanks everybody.

Commit:
bg color added to hero
